tests/cpp/dlr_hexagon/dlr_hexagon_test.cc

```diff
@@ -0,0 +1,201 @@
+  std::string line;
+  std::ifstream fp(img_path);
+  uint8_t* img = new uint8_t[size];
+  size_t i = 0;
+  if (fp.is_open()) {
+    while (getline(fp, line) && i < size) {
+      int v = std::stoi(line);
+      img[i++] = v;
+    }
+    fp.close();
+  }
+
+  EXPECT_EQ(size, i);
+  LOG(INFO) << "Image read - OK, uint8_t[" << i << "]";
+  return img;
+}
+
+int ArgMax(uint8_t* data, int size) {
+  int idx = 0;
+  uint8_t v = 0;
+  for (int i = 0; i < size; i++) {
+    uint8_t vi = data[i];
+    if (vi > v) {
+      idx = i;
+      v = vi;
+    }
+  }
+  return idx;
+}
+
+void CheckAllDLRMethods(DLRModelHandle& handle) {
+  // GetDLRBackend
+  const char* backend_name;
+  if (GetDLRBackend(&handle, &backend_name)) {
+    FAIL() << "GetDLRBackend failed";
+  }
+  LOG(INFO) << "GetDLRBackend: " << backend_name;
+  EXPECT_STREQ("hexagon", backend_name);
+
+  // GetDLRNumInputs
+  int num_inputs;
+  if (GetDLRNumInputs(&handle, &num_inputs)) {
+    FAIL() << "GetDLRNumInputs failed";
+  }
+  LOG(INFO) << "GetDLRNumInputs: " << num_inputs;
+  EXPECT_EQ(1, num_inputs);
+
+  // GetDLRNumOutputs
+  int num_outputs;
+  if (GetDLRNumOutputs(&handle, &num_outputs)) {
+    FAIL() << "GetDLRNumOutputs failed";
+  }
+  LOG(INFO) << "GetDLRNumOutputs: " << num_outputs;
+  EXPECT_EQ(1, num_outputs);
+
+  // GetDLRNumWeights
+  int num_weights;
+  if (GetDLRNumWeights(&handle, &num_weights)) {
+    FAIL() << "GetDLRNumWeights failed";
+  }
+  LOG(INFO) << "GetDLRNumWeights: " << num_weights;
+  EXPECT_EQ(0, num_weights);
+
+  // GetDLRInputName
+  const char* input_name;
+  if (GetDLRInputName(&handle, 0, &input_name)) {
+    FAIL() << "GetDLRInputName failed";
+  }
+  LOG(INFO) << "DLRInputName: " << input_name;
+  EXPECT_STREQ("input", input_name);
+
+  // GetDLROutputSizeDim
+  int64_t out_size;
+  int out_dim;
+  if (GetDLROutputSizeDim(&handle, 0, &out_size, &out_dim)) {
+    FAIL() << "GetDLROutputSizeDim failed";
+  }
+  LOG(INFO) << "GetDLROutputSizeDim.size: " << out_size;
+  LOG(INFO) << "GetDLROutputSizeDim.dim: " << out_dim;
+  EXPECT_EQ(1001, out_size);
+  EXPECT_EQ(4, out_dim);
+
+  // GetDLROutputShape
+  int64_t shape[out_dim];
+  if (GetDLROutputShape(&handle, 0, shape)) {
+    FAIL() << "GetDLROutputShape failed";
+  }
+  std::stringstream ss;
+  ss << "GetDLROutputShape: (" << shape[0];
+  for (int i = 1; i < out_dim; i++) {
+    ss << "," << shape[i];
+  }
+  ss << ")";
+  LOG(INFO) << ss.str();
+  const int64_t exp_shape[4] = {1, 1, 1, 1001};
+  EXPECT_TRUE(std::equal(std::begin(exp_shape), std::end(exp_shape), shape));
+
+  // Load image
+  size_t img_size = 224 * 224 * 3;
+  uint8_t* img = LoadImageAndPreprocess("cat224-3.txt", img_size);
+  LOG(INFO) << "Input sample: [" << +img[0] << "," << +img[1] << "..."
+            << +img[img_size - 1] << "]";
+
+  // SetDLRInput
+  const int64_t in_shape[4] = {1, 224, 224, 3};
+  if (SetDLRInput(&handle, input_name, in_shape, (float*)img, 4)) {
```

Comment:
DLR API uses `float*` pointer to get starting address of image buffer. We still can pass starting address of uint8_t image buffer to DLR by casting `uint8_t*` to `float*`. Pointer size does not depend of data type. It just contains memory address - 64 or 32 bits depending on the platform.  It is ok to cast pointers in this case.